### DISCUSSION TITLE
Convert the MQTT broker response to SOS using the FROST-Server

### DISCUSSION THREAD

**Discussion by malnajdi** [FraunhoferIOSB/FROST-Server#640] [Q&A] [answered]
Hey everyone, 

I want to ask if this is possible of not, i have an existing MQTT broker and i was looking for a way to change the MQTT response to a Sensor Observation Service (SOS) compliant response using FROST-Server. If this is possible please guide me to it. Thanks a lot.

**Comment by hylkevds** [comment, marked as answer]
It should not be very complicated to create a small program that:

1. subscribes to v1.1/Observations
2. converts each received message in into some other format (Fetching any missing data using the SensorThings API)
3. publishes this new format on a different topic (on the same, or another broker)

But it's not functionality that should be part of FROST-Server itself.

**Reply by malnajdi** [reply]
hmmm to be honest i did not understand how can i convert an already existing mqtt broker topic --> SOS using SROST-Server. 
i will search more.

**Comment by hylkevds** [comment]
I doubt anyone has already created such a bridge, so you'll have to create (code) it yourself.
